components/sensor/collector.go

```diff
@@ -0,0 +1,69 @@
+package sensor
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"go.viam.com/rdk/data"
+)
+
+type method int64
+
+const (
+	readings method = iota
+)
+
+func (m method) String() string {
+	if m == readings {
+		return "Readings"
+	}
+	return "Unknown"
+}
+
+// ReadingRecords a collection of ReadingRecord.
+type ReadingRecords struct {
+	Readings []ReadingRecord
+}
+
+// ReadingRecord a single analog reading.
+type ReadingRecord struct {
+	ReadingName string
+	Reading     interface{}
+}
+
+func newSensorCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	sensorResource, err := assertSensor(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	cFunc := data.CaptureFunc(func(ctx context.Context, arg map[string]*anypb.Any) (interface{}, error) {
+		var records []ReadingRecord
+		values, err := sensorResource.Readings(ctx, nil) // TODO: pass in something here from the config rather than nil?
+		if err != nil {
+			return nil, data.FailedToReadErr(params.ComponentName, readings.String(), err)
+		}
+		for name, value := range values {
+			if len(arg) != 0 {
+				// If specific sensor reading names were passed in the robot config, report only those
```

Comment:
Neat, I like this functionality. I want to be consistent in how we're passing in reader names between the sensor and board components. Do you think a user will typically want to just grab _all_ readers?

If so, I'll update the board collector similarly — if `len(arg) == 0`, grab all `AnalogReaderNames()` and create a reader per name. cc @npmenard who added the board collector.

Reply:
I do indeed think a user will generally want to grab all readers; I certainly did. IMO having the "Data Capture Configuration" field attached to a component means _a priori_ "I want data from this component", and if the user cares enough about specifics and granularity then they can add that too. If I add another analog or digital reader to my board I'll likely want that captured too.

Also in the case where there's one analog or one digital reader, and the board has a data capture config of any kind, that's pretty clear what's desired.

Reply:
Though I would agree that the structure between sensor and board should be identical here; so if it's decided not to update board then this should change to mimic that.

Reply:
I would keep this functionality as-is, and I created https://viam.atlassian.net/browse/DATA-1104 to update accordingly in board.